test/controllers/tasks_controller_test.rb

```diff
@@ -90,37 +81,108 @@
   # Unskip and complete these tests for Wave 3
   describe "edit" do
     it "can get the edit page for an existing task" do
-      skip
-      # Your code here
+      # Act
+      get edit_task_path(task.id)
+
+      # Assert
+      must_respond_with :success
     end
 
     it "will respond with redirect when attempting to edit a nonexistant task" do
-      skip
-      # Your code here
+      # Act
+      get edit_task_path(-1)
+
+      # Assert
+      must_respond_with :redirect
     end
   end
 
   # Uncomment and complete these tests for Wave 3
+
   describe "update" do
     # Note:  If there was a way to fail to save the changes to a task, that would be a great
     #        thing to test.
     it "can update an existing task" do
       # Your code here
+      existing_task = Task.create(name: "task_90", description: "prepare my lunch", completed: nil)
+      updated_task_form_data = {
+        task: {
+          name: "task_80",
+          description: "prepare my lunch",
+          completed: nil,
+        }
+      }
+      
+      patch task_path(existing_task.id), params: updated_task_form_data
+
+      expect(Task.find_by(id: existing_task.id).name).must_equal "task_80"
     end
 
-    it "will redirect to the root page if given an invalid id" do
+    it "can't update an existing task giving wrong parameters" do
       # Your code here
+      existing_task = Task.create(name: "task_100", description: "clean the house", completed: nil)
+      updated_task_form_data = {
+        task: {
+          names: "task_70",
+          description: "clean the house",
+          completed: nil,
+        } 
+      }
+      
+      patch task_path(existing_task.id), params: updated_task_form_data
+
+      expect(Task.find_by(id: existing_task.id).name).must_equal nil
```

Comment:
This throws a warning!

`.DEPRECATED: Use assert_nil if expecting nil from test/controllers/tasks_controller_test.rb:134. This will fail in Minitest 6.
.......DEPRECATED: Use assert_nil if expecting nil from test/controllers/tasks_controller_test.rb:74. This will fail in Minitest 6.`